src/lib/offline-api/index.ts

```diff
@@ -341,6 +363,46 @@ class OfflineAPI {
 
     // Mutate version bump
     entry.version = entry.version + 1
+    entry.isPublished = true
+
+    return entry
+  }
+
+  async unpublishEntry (id: string): Promise<Entry> {
+    this.assertRecording()
+
+    const hasEntry = this.entries.some((entry) => entry.id === id)
+
+    if (!hasEntry) {
+      throw new Error(`Cannot unpublish Entry ${id} because it does not exist`)
+    }
+
+    const entry = this.entries.find((entry) => entry.id === id)
+
+    // Store clone as a request
+    this.currentRequestsRecorded.push(unpublishEntryRequest(entry.clone()))
```

Comment:
In this case we can skip the clone since both requests do not require the entry payload, and `id` and `version` cannot be mutated once the request objects have been created

Reply:
You are correct. However it seems that OfflineApi always clones the entry before creating the requests, so the behavior is consistent with the rest of the code. If we change this, we should change it for all of the requests. What do you think?

Reply:
You're correct. Maybe let's just keep it consistent and always clone like you already did.
The costs are most likely negligible and if anything ever changes we're always on the safe side.